semantic-analysis/src/Analysis/Exception.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+module Analysis.Exception
+( Exception(..)
+, ExcSet(..)
+, fromExceptions
+, var
+, exc
+  -- * Exception tracing analysis
+, ExcC(..)
+) where
+
+import           Analysis.Effect.Domain
+import           Analysis.Name
+import           Control.Algebra
+import           Control.Applicative (Alternative (..))
+import qualified Data.Foldable as Foldable
+import qualified Data.Set as Set
+
+newtype Exception = Exception { exceptionName :: Name }
```

Comment:
Can we add a docstring here, so that editor support makes it clear that this is not the `Exception` from `Control.Exception`?

Reply:
Excellent point 👍🏻